src/serializer/serializer.ts

```diff
@@ -14,25 +14,30 @@ import type {
   MappedTypeNode,
   ObjectExpressionNode,
   PropertyNode,
+  RuntimeEnumDeclarationNode,
   StatementNode,
   UnionExpressionNode,
 } from '../ast';
 import { NodeType } from '../ast';
+import { toCamelCase } from '../transformer';
 
 const IDENTIFIER_REGEXP = /^[$A-Z_a-z][\w$]*$/;
 
 export type SerializerOptions = {
   typeOnlyImports?: boolean;
+  camelCase?: boolean;
 };
 
 /**
  * Creates a TypeScript output string from a codegen AST.
  */
 export class Serializer {
   readonly typeOnlyImports: boolean;
+  readonly camelCase: boolean;
 
   constructor(options: SerializerOptions = {}) {
     this.typeOnlyImports = options.typeOnlyImports ?? true;
+    this.camelCase = options.camelCase ?? false;
```

Comment:
I didn't like passing camelCase to this layer. Arguably setting the name for enum values should be handled by the transformer, and we should have a special RuntimeEnumNode instead of a UnionExpressionNode containing string literals. That would also allow us to remove some type assertions in this file. I was just trying to keep the overhead on this PR small, but happy to iterate.